### DISCUSSION TITLE
Saving DWG file produces error

### DISCUSSION THREAD

**Discussion by Spencer-reborn-Elliott** [DomCR/ACadSharp#256] [Q&A] [answered]
I wrote a very small piece of test code to load a DWG file and then without editting it, save it to a specified location. I wanted to make sure that no edits I was making could be causing the issue. 

I am making a windows forms application using .NET 8.0 and nuget package ACadSharp version 2.0.2-beta

The code I used:

`        private void SaveBtn_Click(object sender, EventArgs e)
        {
            CadDocument doc = DwgReader.Read(selectFile.FileName);
            try
            {
                DwgWriter.Write(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "output.dwg"), doc);
            }
            catch (Exception ex)
            {
                LogIt(ex.Message);
            }
        }`

The LogIt function is basically just writing whatever string I pass to a rich text box in the application; In this case it is passing the error message I am catching when I try to save the file. 

The error I receive is:

> Object not implemented : ACadSharp.IO.Templates.DwgColorTemplate+DwgColor

Is there anything I can do to solve this myself? All of the DWG files I am testing produce this same error.

**Comment by DomCR** [comment]
Hi @Spencer-reborn-Elliott,

I've found an error reading the file where is reading an invalid object, I'll open an issue to fix it and make a release.

Thanks!

**Reply by DomCR** [reply]
- https://github.com/DomCR/ACadSharp/pull/257

**Reply by Spencer-reborn-Elliott** [reply]
@DomCR Thank you very much for such a quick reply. I updated the nuget package to 2.0.3-beta and the original error is no-longer present!
Sadly, a new error is present using the exact same conditions as previous (other than the updated nuget package). This error states:

> Object not implemented : ACadSharp.Tables.MultiLeaderStyle

I'm sorry for the bother.

**Reply by DomCR** [reply, marked as answer]
Hi @Spencer-reborn-Elliott,

I'll check the objects for the writer, I see that some of them should be ignored due a lack of implementation, I'll make the changes.

Thanks to report this issue.

**Comment by Spencer-reborn-Elliott** [comment]
> Hi @Spencer-reborn-Elliott,
> 
> I'll check the objects for the writer, I see that some of them should be ignored due a lack of implementation, I'll make the changes.
> 
> Thanks to report this issue.


Thank you very much! The library is super quick at loading the CAD files, really glad I found this - Great work!

**Comment by Spencer-reborn-Elliott** [comment]
> Hi @Spencer-reborn-Elliott,
> 
> I'll check the objects for the writer, I see that some of them should be ignored due a lack of implementation, I'll make the changes.
> 
> Thanks to report this issue.

The nuget package released today with the updates applied fixed the saving issue! Thank you very much!
